tests/gen/gen_auto.go

```diff
@@ -0,0 +1,246 @@
+// Copyright 2023 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package gen
+
+import (
+	"math/rand"
+
+	"github.com/sourcenetwork/defradb/client"
+	"github.com/sourcenetwork/defradb/client/request"
+)
+
+const (
+	defaultNumDocs           = 10
+	defaultNumChildrenPerDoc = 2
+
+	defaultStrLen = 10
+	defaultIntMin = 0
+	defaultIntMax = 10000
+)
+
+type docRec struct {
+	doc    map[string]any
+	docKey string
+}
+
+// AutoGenerateFromSchema generates random documents from a schema.
+func AutoGenerateFromSchema(schema string, options ...Option) ([]GeneratedDoc, error) {
```

Comment:
todo: I think `schema` is a misleading term here, this function takes a GQL SDL document that may contain many schema (and collection) definitions. `Schema` is too easily confused with actual defra `Schema`, and it is made worse by the param being of type `string` (is it json of a single defra schema?  Even reading the function source code it is not possible to tell given the (sensible) abstractions used).

I think this should be renamed to something else. Maybe `AutoGenerateFromGQLSDL`, although the two acronyms blend together with that, maybe `AutoGenerateFromSDL` is sufficient.